gpflow/kernels/misc.py

```diff
@@ -139,10 +144,19 @@ def K(self, X, X2=None):
             X2 = X
         else:
             X2 = tf.cast(X2[:, 0], tf.int32)
+
+        shape_constraints = [
+            (X, [..., 'N', 1]),
+            (X2, [..., 'M', 1]),
+        ]
+        tf.debugging.assert_shapes(shape_constraints)
```

Comment:
I prefer having argument validation as the very first thing in a function, partly so that feedback can be given on the exact data the caller provided, and partly as a good practice of catching errors early

Reply:
Won't work because X2 may be None though